### ISSUE TITLE
JVM does not shutdown if BioImage are created in non-main thread

### ISSUE THREAD

**Issue by lukaslcf** [bioio-devs/bioio-bioformats#23]
### Describe the Bug

I am using Bioio-bioformats to read .lif files (to circumvent the `readlif` bug) inside my PySide6 program. I have a button programmed to open the file when pressed. Since this is a gui program, actions like responding to a button press is handled in a separate thread. Files are read properly and things works. However, when I close the gui window, the app get stuck and I need to manually stop the program by doing Ctrl+C .

I also checked the exit code from the Qt Window to make sure it is returning normally, and it is. But it hangs at the code calling `sys.exit()`.

I tried the solution mentioned in https://github.com/jpype-project/jpype/issues/1169.

I called `jpype.java.lang.Thread.detach()` and `jpype.java.lang.Thread.attachAsDaemon()` just before I call `BioImage()`.

After doing this, the program now exits successfully.

The reasoning as mentioned in the link is that if `scyjava` or `jpype`'s `startJVM()` is being run in a non-main thread, it will attach to that thread and can only be stopped by explicitly calling `shutdownJVM()`, and this call can only be done on the same thread as where `startJVM()` was called.

### Expected Behavior

Expecting that once the `BioImage` object is no longer referenced, it should stop the java instance or at least detach the java instance from the running thread, so that the JVM can be stopped when the python program exits.

### Reproduction

``` Python
import threading

def main():
    from bioio import BioImage
    import bioio_bioformats
    image = BioImage("image.lif", reader=bioio_bioformats.Reader)
    print(image.scene)

t = threading.Thread(target=main, name='jvm-starter')
t.start()
t.join()

```

This code will not exit.

### Environment

-   OS Version: Windows 11
-   bioio-bioformats Version: 1.0.1

**Comment by SeanLeRoy** [comment]
This is interesting! It might be hard to address this within `bioio` itself since we don't keep track of the context of the readers too well - but I'll bring this up to some of the other developers and see if we can brainstorm any quick wins.

**Comment by Thrameos** [comment]
Does the same thing happen if the thread is launched using the Java API as a daemon thread?
